docs/javascript-interop.md

```diff
@@ -23,35 +23,49 @@ A Blazor app can invoke JavaScript functions from .NET and .NET methods from Jav
+To call into JavaScript from .NET, use the `IJSRuntime` abstraction, which is accessible from `JSRuntime.Current`. The `InvokeAsync<T>` method on `IJSRuntime` takes an identifier for the JavaScript function you wish to invoke along with any number of JSON-serializable arguments. The function identifier is relative to the global scope (`window`). If you wish to call `window.someScope.someFunction`, the identifier is `someScope.someFunction`. There's no need to register the function before it's called. The return type `T` must also be JSON serializable.
 
-*exampleJsInterop.js*:
+In the sample app, two JavaScript functions are available to the client-side app that interact with the DOM to receive user input and display a welcome message:
 
-```javascript
-window.exampleJsFunctions = {
-  showPrompt: function (message) {
-    return prompt(message, 'Type anything here');
-  }
-};
-```
+* `showPrompt` &ndash; Produces a prompt to accept user input (the user's name) and returns the name to the caller.
+* `displayWelcome` &ndash; Assigns a welcome message from the caller to a DOM object with an `id` of `welcome`.
 
-*ExampleJsInterop.cs*:
+*wwwroot/exampleJsInterop.js*:
 
-```csharp
-public class ExampleJsInterop
-{
-    public static Task<string> Prompt(string message)
-    {
-        // Implemented in exampleJsInterop.js
-        return JSRuntime.Current.InvokeAsync<string>(
-            "exampleJsFunctions.showPrompt",
-            message);
-    }
-}
-```
+[!code-javascript[](./common/samples/2.x/BlazorSample/wwwroot/exampleJsInterop.js?highlight=2-7)]
+
+Place the `<script>` tag that references the JavaScript file in the *wwwroot/index.html* file:
+
+[!code-html[](./common/samples/2.x/BlazorSample/wwwroot/index.html?highlight=16)]
+
+Don't place a script tag in a component file because the script tag can't be updated dynamically. For more information, see [Add compiler error if there's a &lt;script&gt; element inside a component (aspnet/Blazor &num;552)](https://go.microsoft.com/fwlink/?linkid=872131).
+
+.NET methods interop with the JavaScript functions by calling `InvokeAsync<T>` method on `IJSRuntime`.
+
+The sample app uses a pair of C# methods, `Prompt` and `Display`, to invoke the `showPrompt` and `displayWelcome` JavaScript functions:
+
+*JsInteropClasses/ExampleJsInterop.cs*:
+
+[!code-csharp[](./common/samples/2.x/BlazorSample/JsInteropClasses/ExampleJsInterop.cs?name=snippet1&highlight=6-8,14-16)]
 
 The `IJSRuntime` abstraction is asynchronous to allow for out-of-process scenarios. If the app runs in-process and you want to invoke a JavaScript function synchronously, downcast to `IJSInProcessRuntime` and call `Invoke<T>` instead. We recommend that most JavaScript interop libraries use the async APIs to ensure the libraries are available in all Blazor scenarios, client-side or server-side.
```

Comment:
It looks like this text was unchanged, but the PR says it's supposed to also address #257.  I would change out-of-process to server-side and in-process to client-side.